# HG changeset patch
# User Rik <rik@octave.org>
# Date 1355938568 28800
#      Wed Dec 19 09:36:08 2012 -0800
# Node ID 1e9a6285acc46cf68168725b0b08e38b8475052e
# Parent  142de7308ebf486d0ccc148c632b09025866cfa2
Fix segfault with clear -regexp (bug #37924)

* liboctave/util/regexp.cc(ismatch): Check size of match list (> 0)
to determine if there was a match.  Don't index into non-existent
match element.

diff --git a/liboctave/util/regexp.cc b/liboctave/util/regexp.cc
--- a/liboctave/util/regexp.cc
+++ b/liboctave/util/regexp.cc
@@ -421,21 +421,17 @@ regexp::match (const std::string& buffer
   return retval;
 }
 
 bool
 regexp::is_match (const std::string& buffer)
 {
   regexp::match_data rx_lst = match (buffer);
 
-  regexp::match_data::const_iterator p = rx_lst.begin ();
-
-  std::string match_string = p->match_string ();
-
-  return ! match_string.empty ();
+  return rx_lst.size () > 0;
 }
 
 Array<bool>
 regexp::is_match (const string_vector& buffer)
 {
   octave_idx_type len = buffer.length ();
 
   Array<bool> retval (dim_vector (len, 1));
